File: src/alphalib.jl
# Returns the header containing all instructions
# to use alphalib
alphalib_funcs = [
        "alphaprintint",
        "alphaprintstr",
        "alphascan",
    ]

function alphalib_head()
    s = ""
    for f in alphalib_funcs
        s *= "extern $f\n"
    end

    return s
end

File: alphalib/src/alphalib.c
#include <stdio.h>
#include "alphalib.h"

void alphaprintint(num val) {
    printf("%lld", val);
}

void alphaprintstr(const str val) {
    printf("%s", val);
}

num alphascan() {
    num i;
    scanf("%lld", &i);

    return i;
}
